File: js/functions_arrays/reverse.js
// Write a function, reverse that takes a string and returns another string with the characters 
// in reverse order.

function reverse(str) {
    let result = '';
    for (let i = 1; i <= str.length; i++) { // (let i = str.length - 1; i <= 0; i--)
        result += str[str.length-i]; // result += str[i]
    }
    return result;
}

console.log(reverse("fish"))
console.log(reverse('good morning'))

File: ruby/arrays_hashes/reverse.rb
# Ruby String class comes with a `reverse` method that reverses the order of characters in a string:

greeting = "hello world"

# Write a code that gives the same outcome of the reverse method. You can use any method from the Array class.

arr = greeting.split""
arr2 = []

# arr.each.with_index do |n, i|
#     arr2 << arr.slice(-i-1)
# end

# p arr2.join""


arr.each do |n|
    arr2.unshift n
end

p arr2.join""